ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/editor/EditorAgentImpl.java

```diff
@@ -162,25 +164,24 @@ protected void bind() {
             @Override
             public void onResourceEvent(ResourceNodeDeletedEvent event) {
                 ResourceBasedNode node = event.getNode();
-
+                List<EditorPartPresenter> editors = new ArrayList<EditorPartPresenter>(openedEditors.values());
```

Comment:
In this case we need separated collection (or we should use iterator) because we have concurrent Exception. It was main problem in issue IDEX-3329.